main.go

```diff
@@ -110,35 +169,138 @@ func main() {
 	}
 
 	if task, ok := fetchTasks[cmd]; ok {
-		if *jwtFlag {
+		// Get the common fetch options.
+		var fetchOpts commonFetchOptions
+		if cmd == "fetch" {
+			fetchOpts = opts.Fetch.commonFetchOptions
+		} else if cmd == "header" {
+			fetchOpts = opts.Header.commonFetchOptions
+		} else if cmd == "curl" {
+			fetchOpts = opts.Curl.commonFetchOptions
+		}
+
+		// Get the authentication type, with backward compatibility
+		authType := fetchOpts.AuthType
+		if fetchOpts.Jwt {
+			authType = "jwt"
+		}
+		if fetchOpts.Sso {
+			authType = "sso"
+		}
+
+		// Get the credentials file, with backward compatibility
+		credentials := fetchOpts.Credentials
+		if fetchOpts.Json != "" {
+			credentials = fetchOpts.Json
+		}
+
+		scope := fetchOpts.Scope
+		audience := fetchOpts.Audience
+		email := fetchOpts.Email
+		ssocli := fetchOpts.SsoCli
+		setCacheLocation (fetchOpts.Cache)
+
+		// Get the fetch output format, with backward compatibility
+		format := opts.Fetch.Format
+		if opts.Fetch.OldFormat != "" {
+			format = opts.Fetch.OldFormat
+		}
+
+		// Get the curl specific flags
+		curlcli := opts.Curl.CurlCli
+		url := opts.Curl.Url
+
+		if authType == "jwt"{
 			// JWT flow
-			json, err := readJSON(*jsonFile)
+			json, err := readJSON(credentials)
 			if err != nil {
-				fmt.Println("Failed to open file: " + *jsonFile)
+				fmt.Println("Failed to open file: " + credentials)
 				fmt.Println(err.Error())
 				return
 			}
 
+			// Fallback to reading audience from first remaining arg
+			if audience == "" {
+				if len(remainingArgs) > 0 {
+					audience = remainingArgs [0]
+				} else {
+					fmt.Println("Missing audience argument for JWT")
+					return
+				}
+			}
+
 			settings := &sgauth.Settings{
 				CredentialsJSON: json,
-				Audience:        flagSet.Args()[len(flagSet.Args())-1],
+				Audience:        audience,
+			}
+
+			// Prepare taskArgs based on command type
+			var taskArgs []string
+			if cmd == "curl" {
+				taskArgs = append([]string{curlcli, url}, remainingArgs...)
+			} else if cmd == "fetch" {
+				taskArgs = []string{format}
+			}
+			task(settings, taskArgs...)
+		} else if authType == "sso" {
+			// Fallback to reading email from first remaining arg
+			if email == "" {
+				if len(remainingArgs) > 0 {
+					email = remainingArgs [0]
```

Comment:
nit: remove space

Reply:
This is outdated. There should be no spaces before "[" for slice index in the latest commit.